index.bs

```diff
@@ -1088,34 +1104,111 @@ strong map from handle ids to their corresponding objects.
 OwnershipModel = "root" / "none";
 </pre>
 
-<dfn>RemoteReference</dfn> represents a remote reference to an existing ECMAScript
-object in [=handle object map=] in the given [=Realm=].
+<dfn>RemoteReference</dfn> is either a <code>RemoteObjectReference</code>
+representing a remote reference to an existing ECMAScript object in
+[=handle object map=] in the given [=Realm=], or is a <code>SharedReference</code>
+representing a reference to a [=node=] in the given [=Realm=].
 
 <pre class="cddl remote-cddl local-cddl">
 Handle = text;
+SharedId = text;
+</pre>
 
+<pre class="cddl remote-cddl">
 RemoteReference = {
+  RemoteObjectReference //
+  SharedReference
+}
+
+RemoteObjectReference = {
    handle: Handle,
+   ?sharedId: SharedId
+   Extensible
+}
+
+SharedReference = {
+   ?handle: Handle,
+   sharedId: SharedId
    Extensible
 }
 </pre>
 
 Issue: handle "stale object reference" case.
 
+Note: if the provided reference has both <code>handle</code> and
+<code>sharedId</code>, the algorithm will first try to resolve <code>handle</code>,
```

Comment:
Didn't we at least decide that we should use `sharedId` if present, and if that fails then error? I don't think we have a use case for providing an incorrect `sharedId`, whereas sandboxes are one for providing an incorrect `handle` (if there's also a `sharedId`).

Reply:
Implemented this approach.